app/controllers/trips_controller.rb

```diff
@@ -0,0 +1,89 @@
+class TripsController < ApplicationController
+  def index
+    if params[:passenger_id]
+      @passenger = Passenger.find_by(id: params[:passenger_id])
+      @trips = @passenger.trips
+    else
+      @trips = Trip.all
+    end
+  end
+
+  def edit
+    @trip = Trip.find_by(id: params[:id])
+
+    if @trip.nil?
+      head :not_found
+      return
+    end
+  end
+
+  def update
+    @trip = Trip.find_by(id: params[:id])
+
+    if @trip.nil?
+      head :not_found
+      return
+    elsif @trip.update(trip_params)
+      redirect_to trip_path
+      return
+    else
+      render :edit
+      return
+    end
+  end
+
+  def show
+    id = params[:id].to_i
+    @trip = Trip.find_by(id: id)
+
+    if @trip.nil?
+      head :not_found
+      return
+    end
+  end
+
+  def destroy
+    @trip = Trip.find_by(id: params[:id])
+
+    if @trip.nil?
+      head :not_found
+      return
+    else
+      @trip.destroy
+      redirect_to driver_path
```

Comment:
You can also, if this destroyed trip is incomplete, then you can now make the driver available.